inst/tutorials/practical0-1/practical0-1.Rmd

```diff
@@ -0,0 +1,444 @@
+arguments.
+
+This gets even worse when you account for our ability to misspell things:
+
+```{r misspelling, exercise.cap = "Exercise 3", exercise=TRUE}
+misspelled_add <- function(first, second) {
+  resuIt <- first + second
+  result
+}
+
+result <- -5
+misspelled_add(10, 20)
+misspelled_add(1, 2)
+```
+
+<div id="misspelling-hint">
+**Hint:** If you can't see the problem see how the word `result` is spelled in
+different places.
+</div>
+
+You should find that `misspelled_add()` gives the wrong answer even though the
+code looks right (if you don't look too closely). This problem can (and does!)
+cause hours of agony to some students every year on this course. As a result,
+we insist on two things in your submitted practicals:
+
+1. You use variable names inside functions that you don't use outside in your
+   scripts wherever possible. Sometimes there really only is one sensible name
+   for a variable (maybe `timestep` when describing the increments in your
+   simulation model), and you should be very careful when that happens, but
+   normally there are at least two ways of describing the same idea - often one
+   more general in a function (`birth.rate`) which can be used in multiple
+   contexts, and then specific in your main script each time you use it
+   (`human.annual.birth.rate`, for instance).
+2. You write checks for your functions to make sure they aren't accidentally
+   missing any variables or arguments. There are two tools for this,
+   `findGlobals()` and `checkUsage()` in the `codetools` library.
+
+Try adding the following lines to the end of *Exercise 3*:
+
+```{r, eval=FALSE}
+library(codetools)
+checkUsage(misspelled_add)
+```
+
+You'll see that it reports on the problem that you have probably already
+identified. It can also identify other problems, and we recommend it to you.
+However, it doesn't return anything you can use in your code to allow us to
+automatically reject bad functions, so there is a second solution we'll
+use in our code, and we'll show you how to use it in a later exercise, but
+for now just try using this line instead of `checkUsage()` above (the
+`library()` call is the same):
+
+```{r, eval=FALSE}
+findGlobals(misspelled_add, merge = FALSE)
+```
+
+You'll see that it identifies the use of the global variable `result` too
+(if less elegantly). We'll show you how to use this to automatically detect
+problems with variable misnaming in a later exercise, but for now just remember
+that you need to watch out for misspelling very carefully.
+
+*If you're interested, the `findGlobals()` output also tells you why this
+problem exists in R. You'll see that the function `misspelled_add()` needs
+to use three other *global* variables - the functions `{`, `+` and `<-` - to get
+the function to work. As a result of the way R works, functions have to be able
+to see what is going on outside to be able to work at all. Most languages handle
+this more elegantly.*
```

Comment:
your stars are interfering with each other

Reply:
in fact, for these little asides, I was using quotes, like this:

> interesting aside

Reply:
Yes, I think it's okay though - I like the idea of the quoted aside, but it makes the text bigger (and therefore more prominent). I wish there was a way of avoiding that.